jobs/UnitTest/UnitTest.groovy

```diff
@@ -28,59 +28,80 @@ def setTestRepos(test_repos){
-                try{
-                    sh "./build-config/jobs/UnitTest/unit_test.sh ${repo_name}"
-                } catch(error){
-                    throw error
-                } finally{
-                    archiveArtifacts 'xunit-reports/*.xml'
-                    junit 'xunit-reports/'+"${repo_name}.xml"
-
-                    sh '''
-                    ./build-config/build-release-tools/application/parse_test_results.py \
-                    --test-result-file xunit-reports/'''+"${repo_name}"+'''.xml  \
-                    --parameters-file downstream_file
-                    '''
-                    // Use downstream_file to pass environment variables from shell to groovy
-                    int failure_count = 0
-                    if(fileExists ("downstream_file")) {
-                        def props = readProperties file: "downstream_file"
-                        failure_count = "${props.failures}".toInteger()
-                    }
-                    if (failure_count > 0){
-                        error("There are failed test cases")
+        String node_name = shareMethod.occupyAvailableLockedResource(label_name, used_resources)
+        try{
+            node(node_name){
+                deleteDir()
+                dir("build-config"){
+                    checkout scm
+                }
+                unstash "$stash_manifest_name"
+                env.MANIFEST_FILE_PATH = "$stash_manifest_path"
+                timeout(15){
+                    try{
+                        sh "./build-config/jobs/UnitTest/unit_test.sh ${repo_name}"
+                    } finally{
+                        stash name: "$repo_name", includes: 'xunit-reports/*.xml'
+                        junit 'xunit-reports/'+"${repo_name}.xml"
+    
+                        sh '''
+                        ./build-config/build-release-tools/application/parse_test_results.py \
+                        --test-result-file xunit-reports/'''+"${repo_name}"+'''.xml  \
+                        --parameters-file downstream_file
+                        '''
+                        // Use downstream_file to pass environment variables from shell to groovy
+                        int failure_count = 0
+                        if(fileExists ("downstream_file")) {
+                            def props = readProperties file: "downstream_file"
+                            failure_count = "${props.failures}".toInteger()
+                        }
+                        if (failure_count > 0){
+                            error("There are failed test cases")
+                        }
                     }
                 }
             }
+        } finally{
+            used_resources.remove(node_name)
         }
     }
 }
 
+def archiveArtifactsToTarget(target){
+    try{
+        dir("$target"){
+            for(int i=0; i<test_repos.size; i++){
+                def repo_name = test_repos.get(i)
+                unstash "$repo_name"
```

Comment:
per @PengTian0 , the argument of "$repo_name" should align the  name value in line 45 ```stash name: "$repo_name", includes: 'xunit-reports/*.xml'``` . 
it's a coupling between two functions.  just curious, is it possible to create  a class member variable to align the value of these 2 position ?